docs/navigation/activities.md

```diff
@@ -61,31 +73,87 @@ class ShareRoute(
         })
 }
 
-val shareDestination: NavDestination = ActivityDestination<ShareRoute>(
-    // basic intent that is extended with the Intent above
-    intent = Intent(Intent.ACTION_CHOOSER)
-)
-```
-
-Another example which would open a browser with a given `Uri`:
-
-```kotlin
 // route with a data uri extra
 @Parcelize
 class BrowserRoute(
-    uri: Uri,
+    private val uri: Uri,
 ) : ExternalActivityRoute {
     // the returned Intent is filled into the Intent of the destination by calling
     // destinationIntent.fillIn(fillInIntent())
-    override fun fillInIntent() = Intent().setData(uri)
+    override fun fillInIntent() = Intent(Intent.ACTION_VIEW).setData(uri)
 }
+```
+
+## Activity results
 
-val browserDestination: NavDestination = ActivityDestination<BrowserRoute>(
-    // basic intent that is extended with the Intent above
-    intent = Intent(Intent.ACTION_VIEW)
-)
+External SDKs and the framework often provide an `Activity` that is supposed to be used with
+`startActivityForResult`. AndroidX already introduced `ActivityResultContract` to simplify handling
+this and `ActivityNavigator` uses them to also enable starting them from outside the UI layer
+and receiving results there.
+
+To use the API `registerForActivityResult` needs to be called with an instance of the wanted
+`ActivityResultContract`. This needs to happen before `NavigationSetup` is called for the navigator,
+so it is recommended to do this during the construction of the navigator. The method returns an
```

Comment:
> This needs to happen before `NavigationSetup` is called for the navigator,
so it is recommended to do this during the construction of the navigator. 

Isn't it the only way now? `registerForActivityResult` has to be done either in constructor or as `val` property of navigator. Anything else would be late and cause runtime crash.

Reply:
I think there are some ways where you could do the registration between constructor and setup but let's change it:
```suggestion
so it needs to be called during the construction of the navigator. The method returns an
```